src/vs/server/node/webClientServer.ts

```diff
@@ -128,6 +139,38 @@ export class WebClientServer {
 		return serveFile(this._logService, req, res, filePath, headers);
 	}
 
+	/**
+	 * Handle extension resources
+	 */
+	private async _handleExtensionResource(req: http.IncomingMessage, res: http.ServerResponse, parsedUrl: url.UrlWithParsedQuery): Promise<void> {
+		// Strip `/extensionResource/` from the path
+		const normalizedPathname = decodeURIComponent(parsedUrl.pathname!); // support paths that are uri-encoded (e.g. spaces => %20)
+		const path = normalize(normalizedPathname.substr('/extensionResource/'.length));
+
+		const url = URI.parse(path).with({
+			scheme: this._productService.extensionsGallery?.resourceUrlTemplate ? URI.parse(this._productService.extensionsGallery.resourceUrlTemplate).scheme : 'https',
+			authority: path.substring(0, path.indexOf('/')),
```

Comment:
Could the authority be hard-coded to the authority of `productService.extensionsGallery.resourceUrlTemplate`? If I understand this correctly, the current implementation would transform `/extensionResource/example.com` to `example.com` and it might be better to avoid that.

Reply:
It can be hardcoded because the authority is scoped to publisher. For eg it looks like follows:

`https://vscodevim.vscode-unpkg.net/vscodevim/vim/1.21.10/extension/out/extensionWeb.js`

Reply:
are you protecting for private hostnames and IPs? This could open up Server Side Request Forgery (i.e. ability to retrieve credentials from a cloud vendors metadata service).

Reply:
Even though this redirection is for a specific service, added a check to make sure the redirection url is pointing to the expected service